### ISSUE TITLE
QtCreator missing QML modules (empty welcome screen)

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#9629]
On release-15.09 (f9500fcaaee6e32c6ab8242e31f09af16c9b3fd8), I get this:

```
$ /nix/store/y93kl2kpsvqyhdzgkvcws1ahg5qjdvxn-qtcreator-3.4.2/bin/qtcreator
file:///nix/store/y93kl2kpsvqyhdzgkvcws1ahg5qjdvxn-qtcreator-3.4.2/share/qtcreator/welcomescreen/welcomescreen.qml:41:5: Type SideBar unavailable 
         SideBar { 
         ^
file:///nix/store/y93kl2kpsvqyhdzgkvcws1ahg5qjdvxn-qtcreator-3.4.2/share/qtcreator/welcomescreen/widgets/SideBar.qml:33:1: module "QtQuick.Layouts" is not installed 
     import QtQuick.Layouts 1.0 
     ^
qrc:/timeline/MainView.qml:32:1: module "QtQuick.Controls" is not installed
qrc:/timeline/MainView.qml:33:1: module "QtQuick.Controls.Styles" is not installed
qrc:/timeline/MainView.qml:32:1: module "QtQuick.Controls" is not installed
qrc:/timeline/MainView.qml:33:1: module "QtQuick.Controls.Styles" is not installed
QObject::connect: Cannot connect (null)::updateCursorPosition() to QmlProfiler::Internal::QmlProfilerTraceView::updateCursorPosition()
```

QtCreator does start, but the welcome screen is totally blank. Presumably because of the missing QML modules.

CC @ttuegel @akaWolf.

**Comment by bjornfor** [comment]
Installing qt5Full into my profile (```nix-env -f. -iA qt5Full```) does not fix this issue.

**Comment by akaWolf** [comment]
@bjornfor missing dependencies...
```cat `which qtcreator```
?

**Comment by bjornfor** [comment]
```
$ cat /nix/store/y93kl2kpsvqyhdzgkvcws1ahg5qjdvxn-qtcreator-3.4.2/bin/qtcreator
#! /nix/store/548wfw1i43glkx8lkyjmbg59h6127qky-bash-4.3-p39/bin/bash -e
export QT_PLUGIN_PATH=/nix/store/48ykp6igjwqjglpa8gh6h1sfbg4vz2pi-qtbase-5.4.2/lib/qt5/plugins:/nix/store/nnhd0j59vzy49krprmm55vprga9mydvb-qttools-5.4.2/lib/qt5/plugins:/nix/store/ajpcshs8p70q8ykq8b7rp048hzy6b0q6-qtdeclarative-5.4.2/lib/qt5/plugins:/nix/store/r802j86yp9xmqxg01s20yr1392in4www-qtsvg-5.4.2/lib/qt5/plugins:/nix/store/85mzmrzwhp1ca855fxqhhn96dicb7yhy-qtwebkit-5.4.2/lib/qt5/plugins:/nix/store/smm3pswshlf9a205nkijvss0lx6a4z6w-qtlocation-5.4.2/lib/qt5/plugins:/nix/store/y88732is70r0kjkf9bv77bbln6askx6b-qtmultimedia-5.4.2/lib/qt5/plugins:/nix/store/3kra8shar71r9l3mqqkral9qi46b0ii1-qtsensors-5.4.2/lib/qt5/plugins${QT_PLUGIN_PATH:+:}$QT_PLUGIN_PATH
export QML2_IMPORT_PATH=/nix/store/ajpcshs8p70q8ykq8b7rp048hzy6b0q6-qtdeclarative-5.4.2/lib/qt5/qml:/nix/store/85mzmrzwhp1ca855fxqhhn96dicb7yhy-qtwebkit-5.4.2/lib/qt5/qml:/nix/store/smm3pswshlf9a205nkijvss0lx6a4z6w-qtlocation-5.4.2/lib/qt5/qml:/nix/store/y88732is70r0kjkf9bv77bbln6askx6b-qtmultimedia-5.4.2/lib/qt5/qml:/nix/store/3kra8shar71r9l3mqqkral9qi46b0ii1-qtsensors-5.4.2/lib/qt5/qml${QML2_IMPORT_PATH:+:}$QML2_IMPORT_PATH
exec -a "$0" /nix/store/y93kl2kpsvqyhdzgkvcws1ahg5qjdvxn-qtcreator-3.4.2/bin/.qtcreator-wrapped "${extraFlagsArray[@]}" "$@"
```

**Comment by akaWolf** [comment]
@bjornfor it's strange, since there is no qt54.quickcontrols
workaround: ```nix-env -iA nixpkgs.qt54.quickcontrols```

**Comment by akaWolf** [comment]
https://github.com/NixOS/nixpkgs/blob/f9500fcaaee6e32c6ab8242e31f09af16c9b3fd8/pkgs/development/qtcreator/default.nix#L19 contains quickcontrols, so now we need to consult with @ttuegel.

**Comment by ttuegel** [comment]
The wrapper doesn't mention `quickcontrols`, I'm investigating now.

**Comment by ttuegel** [comment]
OK, there's a problem in the environment settings. This is already fixed with Qt 5.5.

@akaWolf As a workaround, you can add
```
addToSearchPath QML2_IMPORT_PATH "${qtLib.quickcontrols}/lib/qt5/qml"
```
to `postInstall` before calling `wrapProgram`.

**Comment by akaWolf** [comment]
@ttuegel ok, but after we need to delete that string (when the problem with env. sett. will be fixed).
